packages/dds/tree/src/simple-tree/core/allowedTypes.ts

```diff
@@ -226,9 +226,18 @@ export type UnannotateImplicitAllowedTypes<T extends ImplicitAnnotatedAllowedTyp
 			? UnannotateAllowedTypesList<[T]>
 			: T extends readonly (AnnotatedAllowedType | LazyItem<TreeNodeSchema>)[]
 				? UnannotateAllowedTypesList<T>
-				: T extends TreeNodeSchema
+				: // This should able to just return `T` here, however doing so breaks generic code in ArrayNode.
+					T extends TreeNodeSchema
 					? T
-					: never;
+					: // Ideally this case would not be hit as it should be impossible.
+						// Due to limitations of TypeScript some generic code does depend on this case,
+						// so return a valid nonspecific schema in this otherwise unreachable case.
+						// This ensures that when this case is hit, the type system can still reason about it.
+						// It also ensures that if bugs cause this to matter in other cases, they will error on the side of restricting the API
```

Comment:
```suggestion
						// It also ensures that if bugs cause this to matter in other cases, they will err on the side of restricting the API
```

unless this does cause an error that I'm not seeing?